pkg/api/login_oauth.go

```diff
@@ -76,28 +76,30 @@ func (hs *HTTPServer) OAuthLogin(ctx *models.ReqContext) response.Response {
 	loginInfo.AuthModule = name
 	provider := hs.SocialService.GetOAuthInfoProvider(name)
 	if provider == nil {
+		pMsg := "OAuth not enabled"
 		hs.handleOAuthLoginError(ctx, loginInfo, LoginError{
 			HttpStatus:    http.StatusNotFound,
-			PublicMessage: "OAuth not enabled",
+			PublicMessage: pMsg,
 		})
-		return nil
+		return response.Error(http.StatusNotFound, pMsg, errors.New(pMsg))
```

Comment:
As discussed with you, I think this ends up calling the same HTTP primitives as `handleOAuthLoginError` is, resulting in:
`server.go:3159: http: superfluous response.WriteHeader call from [github.com/grafana/grafana/pkg/web.(*responseWriter).WriteHeader](http://github.com/grafana/grafana/pkg/web.(*responseWriter).WriteHeader) (response_writer.go:67)`

Maybe we could have `handleOAuthLoginError` calling the login hooks then returning response instead of calling `ctx.Handle`.
@AgnesToulet do we need `ctx.Handle` to be called before running the `loginHooks`?

Reply:
I am a bit hesitant to change this much only a day before the release 🤔 Beside this error in the log, what are other implications of having the double call? Is this still safe enough change to fix root cause temporary, until we can make necessary changes to improve ?